ddl/job_table.go

```diff
@@ -137,6 +154,41 @@ func (d *ddl) getJob(se *sess.Session, tp jobType, filter func(*model.Job) (bool
 	return nil, nil
 }
 
+func hasSysDB(job *model.Job) bool {
+	sNames := job2SchemaNames(job)
+	// TODO: Handle for the name is empty, like ActionCreatePlacementPolicy.
+	for _, name := range sNames {
+		if tidb_util.IsSysDB(name) {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *ddl) filterEndUserDDL(se *sess.Session, job *model.Job) (bool, error) {
```

Comment:
It would be better to be in ddl/ddl.go

Reply:
Yes, and there's too much code in ddl.go. I'm not going to change it here.